src/Compilers/CSharp/Test/Symbol/Symbols/PartialPropertiesTests.cs

```diff
@@ -3017,11 +3017,1028 @@ partial class C
+        }
+
+        [Fact]
+        public void Attributes_Indexer_DisallowedDuplicates()
+        {
+            var source = """
+                using System;
+
+                class Attr : Attribute { }
+
+                partial class C
+                {
+                    [Attr]
+                    public partial int this[[Attr] int x, [Attr] int y] { [Attr] get; [Attr] [param: Attr] set; } // 1, 2, 3 (param)
+
+                    [Attr] // 4
+                    public partial int this[[Attr] int x, [Attr] int y] { [Attr] get => 1; [Attr] [param: Attr] set { } } // 5, 6 (accessor)
+                }
+                """;
+
+            var comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics(
+                // (8,30): error CS0579: Duplicate 'Attr' attribute
+                //     public partial int this[[Attr] int x, [Attr] int y] { [Attr] get; [Attr] [param: Attr] set; } // 1, 2, 3 (param)
+                Diagnostic(ErrorCode.ERR_DuplicateAttribute, "Attr").WithArguments("Attr").WithLocation(8, 30),
+                // (8,44): error CS0579: Duplicate 'Attr' attribute
+                //     public partial int this[[Attr] int x, [Attr] int y] { [Attr] get; [Attr] [param: Attr] set; } // 1, 2, 3 (param)
+                Diagnostic(ErrorCode.ERR_DuplicateAttribute, "Attr").WithArguments("Attr").WithLocation(8, 44),
+                // (8,86): error CS0579: Duplicate 'Attr' attribute
+                //     public partial int this[[Attr] int x, [Attr] int y] { [Attr] get; [Attr] [param: Attr] set; } // 1, 2, 3 (param)
+                Diagnostic(ErrorCode.ERR_DuplicateAttribute, "Attr").WithArguments("Attr").WithLocation(8, 86),
+                // (10,6): error CS0579: Duplicate 'Attr' attribute
+                //     [Attr] // 4
+                Diagnostic(ErrorCode.ERR_DuplicateAttribute, "Attr").WithArguments("Attr").WithLocation(10, 6),
+                // (11,60): error CS0579: Duplicate 'Attr' attribute
+                //     public partial int this[[Attr] int x, [Attr] int y] { [Attr] get => 1; [Attr] [param: Attr] set { } } // 5, 6 (accessor)
+                Diagnostic(ErrorCode.ERR_DuplicateAttribute, "Attr").WithArguments("Attr").WithLocation(11, 60),
+                // (11,77): error CS0579: Duplicate 'Attr' attribute
+                //     public partial int this[[Attr] int x, [Attr] int y] { [Attr] get => 1; [Attr] [param: Attr] set { } } // 5, 6 (accessor)
+                Diagnostic(ErrorCode.ERR_DuplicateAttribute, "Attr").WithArguments("Attr").WithLocation(11, 77));
+
+            var property = (SourcePropertySymbol)comp.GetMember<NamedTypeSymbol>("C").Indexers.Single();
+            AssertEx.Equal(["Attr", "Attr"], property.GetAttributes().ToStrings());
+            AssertEx.Equal(["Attr", "Attr"], property.GetMethod!.GetAttributes().ToStrings());
+            AssertEx.Equal(["Attr", "Attr"], property.GetMethod!.Parameters[0].GetAttributes().ToStrings());
```

Comment:
Should we take a note for @cston to adjust this test?

Reply:
I wasn't sure what would change about this test as a result of Chuck's work.

Reply:
Not sure why github moved my comment here. It was on `Attributes_Property_BackingField` when I made it; we'll want to test that more thoroughly when `field` is a thing.

Reply:
Added a note to #57012 and comment in source.